src/main/java/org/folio/marc4ld/service/ld2marc/resource/field/ControlFieldsBuilder.java

```diff
@@ -12,30 +11,31 @@
 import org.marc4j.marc.MarcFactory;
 
 public class ControlFieldsBuilder {
-  private static final String EMPTY_CF = SPACE.repeat(39);
+  private static final int CF_008_LENGTH = 39;
   private final Map<String, StringBuilder> storage = new LinkedHashMap<>();
 
   public ControlFieldsBuilder() {
-    storage.put(TAG_008, new StringBuilder(EMPTY_CF));
+    storage.put(TAG_008, new StringBuilder(SPACE.repeat(CF_008_LENGTH)));
   }
 
   public void addFieldValue(String tag, String newValue, Integer startPos, Integer endPos) {
     if (isNotEmpty(newValue)) {
-      storage.putIfAbsent(tag, new StringBuilder(EMPTY_CF));
-      storage.get(tag).replace(startPos, endPos, ensureLength(newValue, endPos - startPos));
+      storage.putIfAbsent(tag, new StringBuilder());
+      newValue = limitLength(newValue, endPos);
```

Comment:
This should be 
`newValue = limitLength(newValue, (endPos - startPos));`
right?

Reply:
That's correct!
Thank you, fixed.